src/main/java/org/candlepin/auth/AnonymousCloudRegistrationAuth.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.auth;
+
+import org.candlepin.config.ConfigProperties;
+import org.candlepin.config.Configuration;
+import org.candlepin.config.ConversionException;
+import org.candlepin.model.AnonymousCloudConsumer;
+import org.candlepin.model.AnonymousCloudConsumerCurator;
+import org.candlepin.pki.CertificateReader;
+import org.candlepin.resteasy.filter.AuthUtil;
+
+import org.jboss.resteasy.spi.HttpRequest;
+import org.keycloak.TokenVerifier;
+import org.keycloak.common.VerificationException;
+import org.keycloak.representations.JsonWebToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.security.PublicKey;
+import java.security.cert.X509Certificate;
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+
+
+/**
+ * AuthenticationProvider that authenticates anonymous access tokens generated from the CloudRegistration endpoint
+ */
+public class AnonymousCloudRegistrationAuth implements AuthProvider {
```

Comment:
This whole class should have probably been written as part of CANDLEPIN-625, but since you've already started it it's fine. Just anote for that task: for any new AuthProvider to work, i needs to be registered in `AuthenticationFilter.setupAuthStrategies()`.